packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -333,12 +333,22 @@ async function migrateToArmAndMultiEnv(ctx: CoreHookContext): Promise<void> {
   await postMigration(projectPath, ctx, inputs, backupFolder);
 }
 
+async function getManifestPath(fx: string, projectPath: string): Promise<string> {
+  // 2.0.1 <= version <= 2.3.1
+  if (await fs.pathExists(path.join(fx, REMOTE_MANIFEST))) {
+    return path.join(fx, REMOTE_MANIFEST);
+  }
+  // 2.3.2<= version <= 2.4.1
+  if (await fs.pathExists(path.join(fx, AppPackageFolderName, REMOTE_MANIFEST))) {
+    return path.join(fx, AppPackageFolderName, REMOTE_MANIFEST);
+  }
+  return path.join(projectPath, AppPackageFolderName, REMOTE_MANIFEST);
+}
+
 async function preCheckKeyFiles(projectPath: string, ctx: CoreHookContext): Promise<void> {
   const core = ctx.self as FxCore;
   const fx = path.join(projectPath, `.${ConfigFolderName}`);
-  const manifestPath = (await fs.pathExists(path.join(fx, AppPackageFolderName, REMOTE_MANIFEST)))
-    ? path.join(fx, AppPackageFolderName, REMOTE_MANIFEST)
-    : path.join(projectPath, AppPackageFolderName, REMOTE_MANIFEST);
+  const manifestPath = await getManifestPath(fx, projectPath);
   await preReadJsonFile(path.join(fx, "env.default.json"), core);
   await preReadJsonFile(path.join(fx, "settings.json"), core);
   await preReadJsonFile(manifestPath, core);
```

Comment:
@XiaofuHuang if manifestPath is invalid, will be caught here. Not only manifest path not found, but also format error, permission error.